File: Client/BookClient/src/app/_models/book.ts
import { Photo } from './photo';

export interface Book {
    id: number;
    title: string;
    subTitle: string;
    description: string;
    authorId: number;
    authorName: string;
    publishedDate: Date;
    photoUrl: string;
    publisher?: string;
    photos?: Photo[];
  }

File: BookShop.Domain/Entities/Book.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Text;

namespace BookShop.Domain.Entities
{
    public class Book : AbstractEntity
    {       
        public int Id { get; set; }
        [Required]
        [MaxLength(150)]
        public string Title { get; set; }
        [Required]
        [MaxLength(150)]
        public string Subtitle { get; set; }
        [MaxLength(2500)]
        public string Description { get; set; }
        public int AuthorId { get; set; }
        public Author Author { get; set; }
        public DateTime? publishedDate { get; set; }
        public string publisher { get; set; }
        public ICollection<Photo> Photos { get; set; }
    }
}
